src/Calculator/SettingsPage.xaml.cpp

```diff
@@ -0,0 +1,177 @@
+    }
+}
+
+void SettingsPage::InitializeContributeTextBlock()
+{
+    std::wstring contributeHyperlinkText  = resourceLoader->GetResourceString(L"SettingsContribute")->Data();
+
+    // The resource string has '%HL%' wrapped around 'GitHub'
+    // Break the string and assign pieces appropriately.
+    static const std::wstring delimiter{ L"%HL%" };
+    static const size_t delimiterLength{ delimiter.length() };
+
+    // Find the delimiters.
+    size_t firstSplitPosition = contributeHyperlinkText .find(delimiter, 0);
+    assert(firstSplitPosition != std::wstring::npos);
+    size_t secondSplitPosition = contributeHyperlinkText .find(delimiter, firstSplitPosition + 1);
+    assert(secondSplitPosition != std::wstring::npos);
+    size_t hyperlinkTextLength = secondSplitPosition - (firstSplitPosition + delimiterLength);
+
+    // Assign pieces.
+    auto contributeTextBeforeHyperlink = ref new String(contributeHyperlinkText.substr(0, firstSplitPosition).c_str());
+    auto contributeTextLink = ref new String(contributeHyperlinkText.substr(firstSplitPosition + delimiterLength, hyperlinkTextLength).c_str());
+    auto contributeTextAfterHyperlink = ref new String(contributeHyperlinkText.substr(secondSplitPosition + delimiterLength).c_str());
+
+    ContributeRunBeforeLink->Text = contributeTextBeforeHyperlink;
+    ContributeRunLink->Text = contributeTextLink;
+    ContributeRunAfterLink->Text = contributeTextAfterHyperlink;
+
+}
+
+void SettingsPage::BackButtonClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    MainPageProperty->CollapseSettings();
+}
+
+void SettingsPage::SetVersionString()
+{
+    PackageVersion version = Package::Current->Id->Version;
+    String ^ appName = AppResourceProvider::GetInstance()->GetResourceString(L"AppName");
+
+    SettingsVersion->Text = appName + L" " + version.Major + L"." + version.Minor + L"." + version.Build + L"." + version.Revision;
+}
+
+void SettingsPage::SetCopyrightString()
+{
+    auto copyrightText = LocalizationStringUtil::GetLocalizedString(resourceLoader->GetResourceString("SettingsCopyright"), StringReference(to_wstring(BUILD_YEAR).c_str()));
+
+    SettingsCopyright->Text = copyrightText;
+}
+
+void SettingsPage::ColorSettingsButtonClicked(_In_ Object ^ sender, _In_ RoutedEventArgs ^ e)
+{
+    Launcher::LaunchUriAsync(ref new Uri("ms-settings:colors"));
+}
+
+void SettingsPage::SettingsFeedbackButtonClick(_In_ Object ^ sender, _In_ RoutedEventArgs ^ e)
+{
+    PackageVersion version = Package::Current->Id->Version;
+    String ^ versionNumber = ref new String(L"Version ");
+    versionNumber = versionNumber + version.Major + version.Minor + version.Build + version.Revision;
+
+    Launcher::LaunchUriAsync(ref new Uri("windows-feedback:?contextid=130&metadata=%7B%22Metadata%22:[%7B%22AppBuild%22:%22" + versionNumber + "%22%7D]%7D"));
+}
+
+void SettingsPage::LightChecked(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
```

Comment:
Can you collapse these into a ThemeChanged method that could handle of these cases in one method? This would reduce the duplicative code.